src/main/java/com/nvidia/spark/rapids/jni/DecimalUtils.java

```diff
@@ -180,29 +182,13 @@ public static Table add128(ColumnView a, ColumnView b, int targetScale) {
     return new Table(add128(a.getNativeView(), b.getNativeView(), targetScale));
   }
 
-  /**
-   * A class to store the result of a cast operation from floating point values to decimals.
-   * <p>
-   * Since the result column may or may not be used regardless of the value of hasFailure, we
-   * need to keep it and let the caller to decide.
-   */
-  public static class CastFloatToDecimalResult {
```

Comment:
This is a huge PR, and a large reason for that is the enforcement of a strict declaration order of inner classes and methods.  I think that should be deferred to a followup since I'm not sure everyone on the team would agree the new order is better.

Once that is removed, this PR should be relatively small once whitespace diffs are ignored.